plugins/modules/ec2_vpc_net.py

```diff
@@ -384,7 +384,7 @@ def wait_for_vpc_ipv6_state(module, connection, vpc_id, ipv6_assoc_state):
     If ipv6_assoc_state is False, wait for VPC to be dissociated from all Amazon-provided IPv6 CIDR blocks.
     """
 
-    if ipv6_assoc_state is None:
+    if not ipv6_assoc_state:
```

Comment:
This change would mean that we don't wait for *dis*associations to complete, which would be the wrong behaviour.  I suspect the bad behaviour is somewhere in the logic around line 410